salt/metalk8s/addons/dex/deployed/secret.sls

```diff
@@ -0,0 +1,110 @@
+    'metalk8s-auth',
+    'metalk8s-dex-config'
+  )
+%}
+
+
+Create Dex Secret:
+  metalk8s_kubernetes.object_present:
+    - manifest:
+        apiVersion: v1
+        kind: Secret
+        metadata:
+          labels:
+            app.kubernetes.io/instance: dex
+            app.kubernetes.io/managed-by: salt
+            app.kubernetes.io/name: dex
+            app.kubernetes.io/part-of: metalk8s
+            app.kubernetes.io/version: 2.19.0
+            helm.sh/chart: dex-2.4.0
+            heritage: metalk8s
+          name: dex
+          namespace: metalk8s-auth
+        stringData:
+          config.yaml: |-
+            issuer: https://{{ grains.metalk8s.control_plane_ip }}:8443/oidc
+            storage:
+              config:
+                inCluster: true
+              type: kubernetes
+            logger:
+              level: debug
+            web:
+              https: 0.0.0.0:5556
+              tlsCert: /etc/dex/tls/https/server/tls.crt
+              tlsKey: /etc/dex/tls/https/server/tls.key
+            {%- if dex.spec.connectors %}
+            connectors:
+              {%- for connector in dex.spec.connectors %}
+
+              {%- if connector['type'] == 'ldap' %}
+                - type: {{ connector['type'] }}
+                  name: {{ connector['name'] }}
+                  id: {{ connector['id'] }}
+                  config:
+                    host: {{ connector['config']['host'] }}
+                    startTLS: {{ connector['config']['startTLS'] }}
+                    rootCAData: {{ connector['config']['rootCAData'] }}
+                    bindDN: {{ connector['config']['bindDN'] }}
+                    bindPW: {{ connector['config']['bindPW'] }}
+                    usernamePrompt: {{ connector['config']['usernamePrompt'] }}
+                    userSearch:
+                      baseDN: {{ connector['config']['userSearch']['baseDN'] }}
+                      filter: {{ connector['config']['userSearch']['filter'] }}
+                      username: {{ connector['config']['userSearch']['username'] }}
+                      idAttr: {{ connector['config']['userSearch']['idAttr'] }}
+                      emailAttr: {{ connector['config']['userSearch']['emailAttr'] }}
+                      nameAttr: {{ connector['config']['userSearch']['nameAttr'] }}
+                    groupSearch:
+                      baseDN: {{ connector['config']['groupSearch']['baseDN'] }}
+                      filter: {{ connector['config']['groupSearch']['filter'] }}
+                      userMatchers:
+                        userAttr: {{ connector['config']['groupSearch']['userMatchers']['userAttr'] }}
+                        groupAttr: {{ connector['config']['groupSearch']['userMatchers']['groupAttr'] }}
+                      nameAttr: {{ connector['config']['groupSearch']['nameAttr'] }}
+
+              {%- endif %}
+
+              {%- endfor %}
+            {%- endif %}
```

Comment:
I don't get why you do this for me you just have to directly use the connectors content

```suggestion
            connectors: {{ dex.spec.connectors | tojson }}
```
And then you don't need to extract secret from the chart

Reply:
True!